doc/zmq.md

```diff
@@ -0,0 +1,116 @@
+# Block and Transaction Broadcasting With ZeroMQ
+
+[ZeroMQ](http://zeromq.org/) is a lightweight wrapper around TCP
+connections, inter-process communications, and shared-memory,
+providing various message-oriented semantics such as publish/subcribe,
+request/reply, and push/pull.
+
+The Bitcoin Core daemon can be configured to act as a trusted "border
+router", implementing the bitcoin wire protocol and relay, making
+consensus decisions, maintaining the local blockchain database,
+broadcasting locally generated transactions into the network, and
+providing a queryable RPC interface to interact on a polled basis for
+requesting blockchain related data.  However, there exists only a
+limited service to notify external software of events like the arrival
+of new blocks or transactions.
+
+The ZeroMQ facility implements a notification interface through a
+set of specific notifiers. Currently there are notifiers that publish
+blocks and transactions. The  supports a set of no binds a "publish"
+port that broadcasts all newly arrived *and validated* blocks and
```

Comment:
This is not actually correct in the current implementation. You'd need to use CValidationInterface if you want that (see an example in rpcmining.cpp for submitblock).

I think it's very important to make this distinction: what does the ZMQ block interface notify you about? Validated blocks or the accepted block chain? The first can include things like orphaned blocks, and does not tell you which block chain we're currently considering active. The second (what is currently implemented) only notifies for changes in the tip, and may skip blocks in reorgs.